staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -165,6 +165,7 @@ func (i *indexedWatchers) terminateAll(objectType reflect.Type, done func(*cache
 // if you set fire time at X, you can get the bookmark within (X-1,X+1) period.
 // This is NOT thread-safe.
 type watcherBookmarkTimeBuckets struct {
+	lock            sync.Mutex
```

Comment:
As comments above, this structure is not thread-safe. Watchers and bookmark buckets are synchronized by mutex in Cacher, can we keep this pattern with current implementation?

Reply:
We should update the comment - this is more tricky now. Previously, it was locked by mutex in Cacher just out-of-the-box. With the changes in this PR, we would need to add locking in one more place, and there is already quite a bit of contention on that lock.
I think this is better approach.

@tedyu - can you please remove the "NOT thread-safe" comment above?